### PR TITLE
Change the default `prog_bar=False` to `True` in `LightningModule.log_grad_norm`

### LINKED ISSUE #11471: Logging grad norm metrics to progress bar by default
## 🐛 Bug

By default, the grad norm metrics are being logged to the progress bar. This spams the logs with too many gradient parameters. 
I observe that the code in https://github.com/PyTorchLightning/pytorch-lightning/blame/9771e7dff6b36f91be259e26f7b2c906749ae3d3/pytorch_lightning/core/lightning.py has the prog_bar flag set to True even though the documentation says its False by default.

    def log_grad_norm(self, grad_norm_dict: Dict[str, float]) -> None:
        """Override this method to change the default behaviour of ``log_grad_norm``.
        If clipping gradients, the gradients will not have been clipped yet.
        Args:
            grad_norm_dict: Dictionary containing current grad norm metrics
        Example::
            # DEFAULT
            def log_grad_norm(self, grad_norm_dict):
                self.log_dict(grad_norm_dict, on_step=False, on_epoch=True, prog_bar=False, logger=True)
        """
        self.log_dict(grad_norm_dict, on_step=True, on_epoch=True, prog_bar=True, logger=True)

### To Reproduce
Current logs look like:
<pre>Epoch 80:  67% 10/15 [02:53&lt;01:18, 15.76s/it, loss=0.429, v_num=l1_0, grad_2.0_norm_model.base_model.conv1.weight_step=2.510, grad_2.0_norm_model.base_model.bn1.weight_step=0.598, grad_2.0_norm_model.base_model.bn1.bias_step=0.519, grad_2.0_norm_model.base_model.layer1.0.conv1.net.weight_step=3.070, grad_2.0_norm_model.base_model.layer1.0.bn1.weight_step=0.484, grad_2.0_norm_model.base_model.layer1.0.bn1.bias_step=0.390, grad_2.0_norm_model.base_model.layer1.0.conv2.weight_step=1.740, grad_2.0_norm_model.base_model.layer1.0.bn2.weight_step=0.266, grad_2.0_norm_model.base_model.layer1.0.bn2.bias_step=0.164, grad_2.0_norm_model.base_model.layer1.1.conv1.net.weight_step=2.170, grad_2.0_norm_model.base_model.layer1.1.bn1.weight_step=0.378, grad_2.0_norm_model.base_model.layer1.1.bn1.bias_step=0.223, grad_2.0_norm_model.base_model.layer1.1.conv2.weight_step=1.450, grad_2.0_norm_model.base_model.layer1.1.bn2.weight_step=0.230, grad_2.0_norm_model.base_model.layer1.1.bn2.bias_step=0.112, grad_2.0_norm_model.base_model.layer2.0.conv1.net.weight_step=2.030, grad_2.0_norm_model.base_model.layer2.0.bn1.weight_step=0.297, grad_2.0_norm_model.base_model.layer2.0.bn1.bias_step=0.211, grad_2.0_norm_model.base_model.layer2.0.conv2.weight_step=1.630, grad_2.0_norm_model.base_model.layer2.0.bn2.weight_step=0.345, grad_2.0_norm_model.base_model.layer2.0.bn2.bias_step=0.219, grad_2.0_norm_model.base_model.layer2.0.downsample.0.weight_step=0.726, grad_2.0_norm_model.base_model.layer2.0.downsample.1.weight_step=0.341, grad_2.0_norm_model.base_model.layer2.0.downsample.1.bias_step=0.219, grad_2.0_norm_model.base_model.layer2.1.conv1.net.weight_step=1.740, grad_2.0_norm_model.base_model.layer2.1.bn1.weight_step=0.271, grad_2.0_norm_model.base_model.layer2.1.bn1.bias_step=0.180, grad_2.0_norm_model.base_model.layer2.1.conv2.weight_step=1.260, grad_2.0_norm_model.base_model.layer2.1.bn2.weight_step=0.306, grad_2.0_norm_model.base_model.layer2.1.bn2.bias_step=0.176, grad_2.0_norm_model.base_model.layer3.0.conv1.net.weight_step=1.420, grad_2.0_norm_model.base_model.layer3.0.bn1.weight_step=0.316, grad_2.0_norm_model.base_model.layer3.0.bn1.bias_step=0.228, grad_2.0_norm_model.base_model.layer3.0.conv2.weight_step=1.750, grad_2.0_norm_model.base_model.layer3.0.bn2.weight_step=0.304, grad_2.0_norm_model.base_model.layer3.0.bn2.bias_step=0.200, grad_2.0_norm_model.base_model.layer3.0.downsample.0.weight_step=0.504, grad_2.0_norm_model.base_model.layer3.0.downsample.1.weight_step=0.232, grad_2.0_norm_model.base_model.layer3.0.downsample.1.bias_step=0.200, grad_2.0_norm_model.base_model.layer3.1.conv1.net.weight_step=1.920, grad_2.0_norm_model.base_model.layer3.1.bn1.weight_step=0.472, grad_2.0_norm_model.base_model.layer3.1.bn1.bias_step=0.243, grad_2.0_norm_model.base_model.layer3.1.conv2.weight_step=1.420, grad_2.0_norm_model.base_model.layer3.1.bn2.weight_step=0.251, grad_2.0_norm_model.base_model.layer3.1.bn2.bias_step=0.153, grad_2.0_norm_model.base_model.layer4.0.conv1.net.weight_step=1.640, grad_2.0_norm_model.base_model.layer4.0.bn1.weight_step=0.456, grad_2.0_norm_model.base_model.layer4.0.bn1.bias_step=0.226, grad_2.0_norm_model.base_model.layer4.0.conv2.weight_step=2.030, grad_2.0_norm_model.base_model.layer4.0.bn2.weight_step=0.195, grad_2.0_norm_model.base_model.layer4.0.bn2.bias_step=0.124, grad_2.0_norm_model.base_model.layer4.0.downsample.0.weight_step=0.790, grad_2.0_norm_model.base_model.layer4.0.downsample.1.weight_step=0.144, grad_2.0_norm_model.base_model.layer4.0.downsample.1.bias_step=0.124, grad_2.0_norm_model.base_model.layer4.1.conv1.net.weight_step=1.690, grad_2.0_norm_model.base_model.layer4.1.bn1.weight_step=0.293, grad_2.0_norm_model.base_model.layer4.1.bn1.bias_step=0.183, grad_2.0_norm_model.base_model.layer4.1.conv2.weight_step=1.760, grad_2.0_norm_model.base_model.layer4.1.bn2.weight_step=0.0969, grad_2.0_norm_model.base_model.layer4.1.bn2.bias_step=0.0264, grad_2.0_norm_model.consensus.attention_pooling_layers.0.weight_step=1.190, grad_2.0_norm_model.consensus.attention_pooling_layers.1.weight_step=4.700, grad_2.0_norm_model.consensus.attention_pooling_layers.2.weight_step=0.901, grad_2.0_norm_model.consensus.attention_encoder_layer.0.weight_step=6.040, grad_2.0_norm_model.consensus.attention_encoder_layer.0.bias_step=0.233, grad_2.0_norm_model.consensus.final_classifiers.weight_step=4.340, grad_2.0_norm_model.consensus.final_classifiers.bias_step=0.479, grad_2.0_norm_total_step=12.00, grad_2.0_norm_model.base_model.conv1.weight_epoch=3.240, grad_2.0_norm_model.base_model.bn1.weight_epoch=0.580, grad_2.0_norm_model.base_model.bn1.bias_epoch=0.297, grad_2.0_norm_model.base_model.layer1.0.conv1.net.weight_epoch=1.910, grad_2.0_norm_model.base_model.layer1.0.bn1.weight_epoch=0.361, grad_2.0_norm_model.base_model.layer1.0.bn1.bias_epoch=0.293, grad_2.0_norm_model.base_model.layer1.0.conv2.weight_epoch=1.400, grad_2.0_norm_model.base_model.layer1.0.bn2.weight_epoch=0.260, grad_2.0_norm_model.base_model.layer1.0.bn2.bias_epoch=0.145, grad_2.0_norm_model.base_model.layer1.1.conv1.net.weight_epoch=1.510, grad_2.0_norm_model.base_model.layer1.1.bn1.weight_epoch=0.251, grad_2.0_norm_model.base_model.layer1.1.bn1.bias_epoch=0.196, grad_2.0_norm_model.base_model.layer1.1.conv2.weight_epoch=1.200, grad_2.0_norm_model.base_model.layer1.1.bn2.weight_epoch=0.206, grad_2.0_norm_model.base_model.layer1.1.bn2.bias_epoch=0.107, grad_2.0_norm_model.base_model.layer2.0.conv1.net.weight_epoch=1.920, grad_2.0_norm_model.base_model.layer2.0.bn1.weight_epoch=0.283, grad_2.0_norm_model.base_model.layer2.0.bn1.bias_epoch=0.237, grad_2.0_norm_model.base_model.layer2.0.conv2.weight_epoch=1.420, grad_2.0_norm_model.base_model.layer2.0.bn2.weight_epoch=0.288, grad_2.0_norm_model.base_model.layer2.0.bn2.bias_epoch=0.202, grad_2.0_norm_model.base_model.layer2.0.downsample.0.weight_epoch=0.626, grad_2.0_norm_model.base_model.layer2.0.downsample.1.weight_epoch=0.266, grad_2.0_norm_model.base_model.layer2.0.downsample.1.bias_epoch=0.202, grad_2.0_norm_model.base_model.layer2.1.conv1.net.weight_epoch=1.590, grad_2.0_norm_model.base_model.layer2.1.bn1.weight_epoch=0.310, grad_2.0_norm_model.base_model.layer2.1.bn1.bias_epoch=0.210, grad_2.0_norm_model.base_model.layer2.1.conv2.weight_epoch=1.110, grad_2.0_norm_model.base_model.layer2.1.bn2.weight_epoch=0.246, grad_2.0_norm_model.base_model.layer2.1.bn2.bias_epoch=0.174, grad_2.0_norm_model.base_model.layer3.0.conv1.net.weight_epoch=1.360, grad_2.0_norm_model.base_model.layer3.0.bn1.weight_epoch=0.322, grad_2.0_norm_model.base_model.layer3.0.bn1.bias_epoch=0.221, grad_2.0_norm_model.base_model.layer3.0.conv2.weight_epoch=1.660, grad_2.0_norm_model.base_model.layer3.0.bn2.weight_epoch=0.322, grad_2.0_norm_model.base_model.layer3.0.bn2.bias_epoch=0.217, grad_2.0_norm_model.base_model.layer3.0.downsample.0.weight_epoch=0.434, grad_2.0_norm_model.base_model.layer3.0.downsample.1.weight_epoch=0.230, grad_2.0_norm_model.base_model.layer3.0.downsample.1.bias_epoch=0.217, grad_2.0_norm_model.base_model.layer3.1.conv1.net.weight_epoch=1.840, grad_2.0_norm_model.base_model.layer3.1.bn1.weight_epoch=0.457, grad_2.0_norm_model.base_model.layer3.1.bn1.bias_epoch=0.248, grad_2.0_norm_model.base_model.layer3.1.conv2.weight_epoch=1.380, grad_2.0_norm_model.base_model.layer3.1.bn2.weight_epoch=0.276, grad_2.0_norm_model.base_model.layer3.1.bn2.bias_epoch=0.162, grad_2.0_norm_model.base_model.layer4.0.conv1.net.weight_epoch=1.590, grad_2.0_norm_model.base_model.layer4.0.bn1.weight_epoch=0.404, grad_2.0_norm_model.base_model.layer4.0.bn1.bias_epoch=0.217, grad_2.0_norm_model.base_model.layer4.0.conv2.weight_epoch=1.980, grad_2.0_norm_model.base_model.layer4.0.bn2.weight_epoch=0.195, grad_2.0_norm_model.base_model.layer4.0.bn2.bias_epoch=0.112, grad_2.0_norm_model.base_model.layer4.0.downsample.0.weight_epoch=0.756, grad_2.0_norm_model.base_model.layer4.0.downsample.1.weight_epoch=0.151, grad_2.0_norm_model.base_model.layer4.0.downsample.1.bias_epoch=0.112, grad_2.0_norm_model.base_model.layer4.1.conv1.net.weight_epoch=1.630, grad_2.0_norm_model.base_model.layer4.1.bn1.weight_epoch=0.281, grad_2.0_norm_model.base_model.layer4.1.bn1.bias_epoch=0.168, grad_2.0_norm_model.base_model.layer4.1.conv2.weight_epoch=1.740, grad_2.0_norm_model.base_model.layer4.1.bn2.weight_epoch=0.128, grad_2.0_norm_model.base_model.layer4.1.bn2.bias_epoch=0.0285, grad_2.0_norm_model.consensus.attention_pooling_layers.0.weight_epoch=0.654, grad_2.0_norm_model.consensus.attention_pooling_layers.1.weight_epoch=5.230, grad_2.0_norm_model.consensus.attention_pooling_layers.2.weight_epoch=0.935, grad_2.0_norm_model.consensus.attention_encoder_layer.0.weight_epoch=3.450, grad_2.0_norm_model.consensus.attention_encoder_layer.0.bias_epoch=0.0967, grad_2.0_norm_model.consensus.final_classifiers.weight_epoch=2.730, grad_2.0_norm_model.consensus.final_classifiers.bias_epoch=0.194, grad_2.0_norm_total_epoch=10.60]on_validation_batch_end</pre>

### Environment

- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @carmocca @edward-io @ananthsub @rohitgr7 @kamil-kaczmarek @Raalsky @Blaizzy

### PR DESCRIPTION
**Pull request by ggare-cmu**
## What does this PR do?

Fixes #11471.

Set the prog_bar and on_step flags to False in accordance with code documentation to avoid unnecessary logging

### Does your PR introduce any breaking changes? If yes, please list them.
Changing prog_bar=False now will stop logging to the progress bar by default, so would be a breaking change for anybody that's using/needs it and now will have to override the method to enable it back.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by akihironitta** [comment]
Hi @ggare-cmu, thank you for sending a fix! Could you update the changelog?

**Comment by ggare-cmu** [comment]
Hi @akihironitta, the changelog has been updated now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -74,6 +74,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Changed
 
+- Set the `prog_bar` and `on_step` flags to False in `log_grad_norm` ([#11472](https://github.com/PyTorchLightning/pytorch-lightning/pull/11472))
```
Comment (awaelchli) [inline CHANGELOG.md:77]:
```suggestion
- Set the `prog_bar` and `on_step` flags to False in `LightningModule.log_grad_norm` ([#11472](https://github.com/PyTorchLightning/pytorch-lightning/pull/11472))
```

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Please also see the comment by @carmocca https://github.com/PyTorchLightning/pytorch-lightning/issues/11471#issuecomment-1013105031
A doc update might be need as well.

**Comment by ggare-cmu** [comment]
@awaelchli updated the PR as recommended. @akihironitta @carmocca @awaelchli Thanks for reviewing!

**Review by carmocca — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Pull request merged**